artifacts.go

```diff
@@ -58,9 +60,30 @@ func artifactWriter(f *os.File, c *cli.Context,
 
 func scripts(c *cli.Context) (*artifact.Scripts, error) {
 	scr := artifact.Scripts{}
-	for _, script := range c.StringSlice("script") {
-		if err := scr.Add(script); err != nil {
-			return nil, err
+	for _, scriptArg := range c.StringSlice("script") {
+		statInfo, err := os.Stat(scriptArg)
+		if err != nil {
+			return nil, errors.Wrapf(err, "can not stat script file: %s", scriptArg)
+		}
+
+		var scriptList []string
+		// Read either a directory, or add the script file directly.
+		if statInfo.IsDir() {
+			fileList, err := ioutil.ReadDir(scriptArg)
+			if err != nil {
+				return nil, errors.Wrapf(err, "can not list directory contents of: %s", scriptArg)
+			}
+			for _, statInfo = range fileList {
+				scriptList = append(scriptList, filepath.Join(scriptArg, statInfo.Name()))
+			}
+		} else {
+			scriptList = []string{scriptArg}
+		}
+
+		for _, script := range scriptList {
```

Comment:
Why are we using separate list here and then we are iterating over and adding to scripts? Can't we do:
```
if statInfo.IsDir() {
  fileList, err := ioutil.ReadDir(scriptArg)
  for _, statInfo = range fileList {
    scr.Add(script)
  } else {
    scr.Add(scriptArg)
  }
```